### PR TITLE
Database connection

### PR DESCRIPTION
**Pull request by TanelOrumaa**
Added database connection capability. In case the database is not local, I've added a possibility to create a SSH tunnel to access it via localhost (with port forwarding). I also created a few example methods for users database layer.
Since this is a public project, I've obviously omitted sensitive data from the commit and will share it with necessary people privately.

### CODE REVIEW

File: src/main/java/database/DatabaseConnection.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+     * Runs input query with this database connection object.
+     * @param query An SQL query with correct syntax.
+     * @return ResultSet object with resulting data.
+     * @throws SQLException If the SQL query is invalid or there's a problem with database connection.
+     */
+    public ResultSet runSelectQuery(String query) throws SQLException {
+        if (this.isActive()) {
+            return this.databaseConnection.createStatement().executeQuery(query);
+        } else {
+            // TODO: Should we create a new connection?
```
Comment (HSellis) [inline src/main/java/database/DatabaseConnection.java]: Probably should, or maybe always check if connection is active before using this method?

File: src/main/java/configuration/Config.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
+package configuration;
+
+import java.io.File;
+
+public class Config {
+
+    // SSH settings
+    public static boolean DATABASE_IN_LOCALHOST = false;
+    public static final String SSH_TUNNE_USERNAME = "obfuscated";
```
Comment (HSellis) [inline src/main/java/configuration/Config.java]: Typo? (L missing from TUNNEL) :)

Comment (TanelOrumaa) [reply src/main/java/configuration/Config.java]: Great find!

File: src/main/java/database/DatabaseConnection.java (hunk 1 above)
Comment (TanelOrumaa) [reply src/main/java/database/DatabaseConnection.java]: I figure we should check statistics in real application and see how often (if ever) this happens and figure out what to do then.

**Pull request merged**